### DISCUSSION TITLE
Make section editing more immersive

### DISCUSSION THREAD

**Discussion by linonetwo** [kookma/TW-Section#15] [Ideas]
About "immersive", I mean make preview mode and edit mode look as similar as possible, so it feels more like a WYSIWYG editor.

1. If combine current editing experience with a floating editor toolbar, there won't be a sudden height change, which will force our eye focus point to move.
2. remove the editor's gray background, instead, make the background the same as the preview mode
3. double click (or even single click, if user is not selecting something) to edit most of the brick, for interactive brick, remain using an edit button (see hypermd for example); click outside of edit area to save and exit the edit mode, besides clicking the save button.

These will make section editing more immersive, and require less operations to edit content, I think.

---

1. Add "add new section" button after a section @Gk0Wk

**Comment by kookma** [comment]
Nice idea!
I would welcome a PR. There is development branch and I think we can experiment there!

**Reply by linonetwo** [reply]
I'm quite busy updating TidGi recently, but may get involved in the future. Right now I may just log down thoughts here~

**Reply by linonetwo** [reply]
And I'm developing a GTD workflow based on what I learned from the talks forum.

**Comment by linonetwo** [comment]
https://github.com/danielo515/TW5-2click2edit/blob/master/2click2editwiki/plugins/2click2edit/clickListener.js  This widget provides double click to edit.

**Reply by kookma** [reply]
I am aware of that and have not used it and prefer to keep more like TW standard!
May be we can have some sub plugins to add such behavior, but for the main plugin I stick with standard TW.
